tuning/trainercontroller/callback.py

```diff
@@ -263,86 +285,102 @@ def on_init_end(
         kwargs["control"] = control
 
         # Check if there any metrics listed in the configuration
-        if CONTROLLER_METRICS_KEY not in self.trainer_controller_config:
+        if (
+            CONTROLLER_METRICS_KEY not in self.trainer_controller_config
+            or len(self.trainer_controller_config[CONTROLLER_METRICS_KEY]) == 0
+        ):
             logger.warning("Trainer controller config has no metrics.")
 
         # Metric handler validation and registration is performed here.
-        for metric_name, metric_config in self.trainer_controller_config[
-            CONTROLLER_METRICS_KEY
-        ].items():
+        for metric_config in self.trainer_controller_config[CONTROLLER_METRICS_KEY]:
+            metric_name = metric_config[CONTROLLER_NAME_KEY]
             # Get the metric class name from the config section.
-            metric_handler_name = next(iter(metric_config.keys()))
+            metric_handler_name = metric_config[CONTROLLER_CLASS_KEY]
             # Get the handler class using the metric class name.
             if metric_handler_name not in self.metric_handlers:
                 raise KeyError(f"Undefined metric handler {metric_handler_name}")
-            metric_handler = self.metric_handlers[metric_handler_name]
+            metric_handler_class = self.metric_handlers[metric_handler_name]
             # Get the metric handler class arguments specified in the config.
-            metric_args = metric_config[metric_handler_name]
-            if metric_args is None:
-                metric_args = {}
+            metric_args = metric_config[ARGS_KEY] if ARGS_KEY in metric_config else {}
             # Metric handler instance is created here.
-            obj = metric_handler(name=metric_name, **metric_args, **kwargs)
+            metric_handler = metric_handler_class(
+                name=metric_name, **metric_args, **kwargs
+            )
             # Add metric instances to the events.
-            for event_name in obj.get_events():
+            for event_name in metric_handler.get_events():
```

Comment:
just a note in general - in the future it would be best to decouple minor changes and refactors from functional changes to keep PRs as small/atomic as possible, as it keeps the PRs easier to review and less likely to be in conflict with each other 🙂

Reply:
Sure